packages/offix-client/src/OfflineClient.ts

```diff
@@ -135,14 +175,22 @@ export class OfflineClient implements ListenerProvider {
    * Restore offline operations into the queue
    */
   protected async restoreOfflineOperations(offlineLink: OfflineLink) {
+    // TODO This is a total mess - needs refactoring
     const offlineMutationHandler = new OfflineMutationsHandler(this.store,
       this.apolloClient as ApolloOfflineClient,
       this.config);
     offlineLink.setup(offlineMutationHandler);
     // Reschedule offline mutations for new client instance
     await offlineMutationHandler.replayOfflineMutations();
+
     // After pushing all online changes check and set network status
-    await offlineLink.initOnlineState();
+
+    // ^^ But why do we wait until now to check and set network status?
+    await offlineLink.initOnlineState(); // TODO this needs to go away
```

Comment:
It will be hard for this to go away because the changes will be schedulerd. 
Offline `this.offlineProcessor.initOnlineState();` can go away as it is being used only as inidicator and do not have any mutations waiting. Hopefuly we can bring all of those outside link at some point and.. then both will go away :D